nixos/lib/make-system-tarball.nix

```diff
@@ -21,14 +21,33 @@
 
   # Extra tar arguments
 , extraArgs ? ""
+  # Method of compression
+, compression ? "pixz"
+  # flags passed to the compressor
+, compressorFlags ? ""
 }:
 
-stdenv.mkDerivation {
+let
+  extensionMap = {
+    xz = ".xz";
+    pixz = ".xz";
+    gzip = ".gz";
+    pigz = ".gz";
+    bzip2 = ".bz2";
+    lzma = ".lzma";
+    none = "";
+  };
+  inputMap = {
+    inherit xz pixz gzip pigz bzip2 lzma;
+    none = null;
+  };
+in stdenv.mkDerivation {
   name = "tarball";
   builder = ./make-system-tarball.sh;
-  buildInputs = [perl xz];
+  buildInputs = [ perl (inputMap.${compression}) ];
 
-  inherit fileName pathsFromGraph extraArgs extraCommands;
+  inherit fileName pathsFromGraph extraArgs extraCommands compressorFlags;
+  compression = if (compression == "none") then "cat" else compression;
```

Comment:
Maybe move this in to the inputMap so there is no special case.

Reply:
`inputMap` is a list of derivations to have in `buildInputs`, but this one is a string, what command to pipe the tar file thru, i would need a 3rd mapping

Reply:
Perhaps instead:

```nix
{
compression = {
  none = {
    inputs = [ ];
    extension = "";
    command = "cat";
  };
  pigz = {
    inputs = [ pigz ];
    extension = ".pigz";
    command = "pigz";
  };
}
}
```
 
?

Reply:
yeah, that can work